initrd/bin/kexec-select-boot

```diff
@@ -50,21 +50,24 @@ bootdir="${bootdir%%/}"
 paramsdev="${paramsdev%%/}"
 paramsdir="${paramsdir%%/}"
 
-PRIMHASH_FILE="$paramsdir/kexec_primhdl_hash.txt"
-if [ "$CONFIG_TPM2_TOOLS" = "y" ]; then
-	if [ -r "$PRIMHASH_FILE" ]; then
-		sha256sum -c "$PRIMHASH_FILE" \
-			|| {
-			echo "FATAL: Hash of TPM2 primary key handle mismatch!";
-			warn "If you have not intentionally regenerated TPM2 primary key,";
-			warn "your system may have been compromised";
-		}
-	else
-		warn "Hash of TPM2 primary key handle does not exist"
-		warn "Please rebuild the boot hash tree"
-		default_failed="y"
-	fi
-fi
+
+#PRIMHASH_FILE="$paramsdir/kexec_primhdl_hash.txt"
+#if [ "$CONFIG_TPM2_TOOLS" = "y" ]; then
+#	if [ -r "$PRIMHASH_FILE" ]; then
+#		sha256sum -c "$PRIMHASH_FILE" \
+#			|| {
+#			echo "FATAL: Hash of TPM2 primary key handle mismatch!";
+#			warn "If you have not intentionally regenerated TPM2 primary key,";
+#			warn "your system may have been compromised";
+#		}
+#	else
+#		warn "Hash of TPM2 primary key handle does not exist"
+#		warn "Please rebuild the boot hash tree"
+#		default_failed="y"
+#	fi
+#fi
+#TODO: Readd when this can work successfully by simply resealing TOTP/HOTP without having to reset TPM2, this is a major pain point for users
```

Comment:
@JonathonHall-Purism this is the part I want you to meditate about.

Reply:
I'll look into this, there are some gaps around the TPM2 primary handle hash that need to be addressed, not sure what the answer to this is yet but I agree this needs some work.

Reply:
Summing up our discussion - at the time of this comment, the prompt was occurring correctly due to swapping qemu boards with the same disk image.  That changes the TPM, which correctly was identified as a mismatched primary handle hash.

We also found that the TPM2 primary handle was moved to /tmp/secret during this PR but kexec-save-default also referenced this file and wasn't updated.

For the purposes of this PR, this change is fine, we fixed the path.

We also discussed the purpose of the primary handle and why it is causing trouble with /boot signatures here:

# Primary handle

The primary handle is essentially a root object in the TPM2 heirarchy.  We use this when sealing, but it could easily be read from the TPM (which we do).

Using the primary handle value to create the session enables parameter transport encryption (prevents a passive attacker from sniffing the bus).  Authenticating the primary handle also prevents an active MITM from intercepting the bus.  This is where the primary handle hash comes into play.

# Primary handle _hash_

The primary handle hash basically authenticates the TPM, preventing any MITM attack on the TPM bus, as the GPG signature on /boot covers the value (via hashes.txt, via the primary handle hash).

Using /boot signatures for this, however, is not ideal.  Resetting the TPM (changing the primary handle) does not update the primary handle hash or re-sign it (as part of /boot), leaving an awkward disconnected workflow.

We _do_ want to authenticate the TPM, and using a GPG signature is a good way to do that.  I think we should consider signing this separately (as part of TPM reset for TPM2).

TPM 1.2 does not have parameter encryption (passive attackers will always be able to sniff the bus).  Any new requirements here, like GPG signing as part of TPM reset, are specific to TPM2 to achieve stronger security guarantees.